File: ollama/thinking.js
import ollama from 'ollama'

async function main() {
  const response = await ollama.chat({
    model: 'qwen3',
    messages: [
      {
        role: 'user',
        content: 'What is bigger, 9.9 or 9.11?',
      },
    ],
    stream: false,
    think: true,
  })

  console.log('Thinking:\n========\n\n' + response.message.thinking)
  console.log('\nResponse:\n========\n\n' + response.message.content + '\n\n')
}

main()

File: ollama/thinking.py
from ollama import chat

messages = [
    {
        "role": "user",
        "content": "What is bigger, 9.9 or 9.11?",
    },
]

response = chat("deepseek-r1", messages=messages, think=True)

print(f"Thinking:\n========\n\n{response.message.thinking or ''}")
print(f"\nResponse:\n========\n\n{response.message.content or ''}")
